packages/catalog-process/src/services/validators.ts

```diff
@@ -100,3 +102,37 @@ export function assertRiskAnalysisIsValidForPublication(
     }
   });
 }
+
+export function assertInterfaceDeletableDescriptorState(
+  descriptor: Descriptor
+): void {
+  match(descriptor.state)
+    .with(descriptorState.draft, () => void 0)
+    .with(
+      descriptorState.archived,
+      descriptorState.deprecated,
+      descriptorState.published,
+      descriptorState.suspended,
+      () => {
+        throw notValidDescriptor(descriptor.id, descriptor.state);
+      }
+    )
+    .exhaustive();
+}
```

Comment:
This was advised by @taglioni-r in a Slack thread. I would start implementing it